tests/test-integration-mock.py

```diff
@@ -0,0 +1,446 @@
+#!/usr/bin/python3
```

Comment:
I afraid I don;t know enough Python (nor umockdev) to give a deep review here, but generally it looks good.